### PR TITLE
Updated button specification with Icon

### PR DESCRIPTION
**Pull request by narefyev91**
### Details

Updated design for Button component with Icon

### Fixed Issues

$ https://github.com/Expensify/App/issues/37089
PROPOSAL: https://github.com/Expensify/App/issues/37089

### Tests
1. Open workspace in which user is an admin
2. Verify that in profile screen - there 2 buttons with icons
3. Verify styles

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Open workspace in which user is an admin
2. Verify that in profile screen - there 2 buttons with icons
3. Verify styles

### QA Steps
1. Open workspace in which user is an admin
2. Verify that in profile screen - there 2 buttons with icons
3. Verify styles

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

android

</details>

<details>
<summary>Android: mWeb Chrome</summary>

android-web

</details>

<details>
<summary>iOS: Native</summary>

ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>

ios-web

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

web

</details>

<details>
<summary>MacOS: Desktop</summary>

desktop

</details>

### CODE REVIEW

**Comment by narefyev91** [comment]
@Expensify/design please review screens with new icon inside button component
web
ios

**Comment by narefyev91** [comment]
> cc @dannymcclain specifically since you made the button component in Figma.
> 
> A couple of questions from me:
> 
> * If you don't set any props on the button component, just confirming that the default button we would use is the "medium" version at 40px tall?
> * I believe we're doing something funky in Figma to use slightly less padding on the side of the button that has the icon. Do we want to do that same treatment here? I believe we are just giving the icon wrapper some negative margin or something like that.

@shawnborton 
1. yup now we have fallback for button 40px.
2. we not having any negative margin now - we just show correct left padding which should be added based on button size and icon (ex from spec)
Screenshot 2024-03-11 at 17 03 56

**Comment by shawnborton** [comment]
Yup, understood, just noting that in Figma we actually are doing something tricky where we are using a wrapper for the icon that is 4px less wide than the icon itself. This way it gives the visual effect of having 4px less padding on the side that has an icon:
CleanShot 2024-03-11 at 11 10 26@2x


So I think ideally we do this here as well. Not sure how we want to implement it - it might be as simple as just writing in the code that when the button has an icon on the left, take 4px of padding off the left side (and do the same for the right).

**Comment by narefyev91** [comment]
> Yup, understood, just noting that in Figma we actually are doing something tricky where we are using a wrapper for the icon that is 4px less wide than the icon itself. This way it gives the visual effect of having 4px less padding on the side that has an icon: CleanShot 2024-03-11 at 11 10 26@2x
> 
> So I think ideally we do this here as well. Not sure how we want to implement it - it might be as simple as just writing in the code that when the button has an icon on the left, take 4px of padding off the left side (and do the same for the right).

Yup. That exactly implemented at that way now - if icon is used - we just adding less padding for left - instead of 12 -> 8, instead of 16 -> 12, instead of 20 -> 16

**Comment by shawnborton** [comment]
Ah okay, that is great - thanks for confirming!

**Review by luacmartins — CHANGES_REQUESTED** [review]
Code LGTM, but I think we need to apply it to the recently merged [tags page](https://github.com/Expensify/App/blob/c39408365326fcf5c69f22f5c8e0a349e21303b7/src/pages/workspace/tags/WorkspaceTagsPage.tsx#L126) too

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Screenshot 2024-03-12 at 16 29 23
Screenshot 2024-03-12 at 16 29 46

Screenshot 2024-03-12 at 16 30 08

Screenshot 2024-03-12 at 16 30 33

</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>


Screenshot 2024-03-12 at 16 38 50
Screenshot 2024-03-12 at 16 39 03
Screenshot 2024-03-12 at 16 39 20

Screenshot 2024-03-12 at 16 40 50

</details>

<details>
<summary>iOS: mWeb Safari</summary>
Screenshot 2024-03-12 at 16 33 43
Screenshot 2024-03-12 at 16 32 49


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-03-12 at 13 18 56
Screenshot 2024-03-12 at 13 18 34
Screenshot 2024-03-12 at 13 18 05


</details>

<details>
<summary>MacOS: Desktop</summary>
Screenshot 2024-03-12 at 16 40 13
Screenshot 2024-03-12 at 16 40 26
Screenshot 2024-03-12 at 16 40 36




</details>

**Comment by sobitneupane** [comment]
|Staging |This PR|
|---|----|
Screenshot 2024-03-12 at 13 18 37 | Screenshot 2024-03-12 at 13 18 34
Screenshot 2024-03-12 at 13 18 08 | Screenshot 2024-03-12 at 13 18 05
Screenshot 2024-03-12 at 13 23 41 | 
Screenshot 2024-03-12 at 13 23 17

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Comment by sobitneupane** [comment]
@luacmartins I have listed few changes https://github.com/Expensify/App/pull/37966#issuecomment-1990957640

**Comment by luacmartins** [comment]
@Expensify/design could you please take a look at these changes? Do they seem correct? https://github.com/Expensify/App/pull/37966#issuecomment-1990957640

**Review by luacmartins — DISMISSED** [review]

**Review by luacmartins — COMMENTED** [review]
Waiting on design approval

**Comment by shawnborton** [comment]
Once we address those comments above, let's do an adhoc build so we can go through and double check buttons as well. Thanks!

**Comment by narefyev91** [comment]
@shawnborton latest main branch
Screenshot 2024-03-12 at 18 29 53

**Comment by shawnborton** [comment]
Ah okay, I was on a Collect workspace with a beta enabled. 

I think given that we're going to deprecate every single page like this:
CleanShot 2024-03-12 at 12 45 14@2x

And do it quite soon, I am not too worried about that particular button style. 

Anyways, @narefyev91 can you address my comment [here](https://github.com/Expensify/App/pull/37966#issuecomment-1991998058), and then I can spin up an adhoc build? Thanks!

**Comment by dannymcclain** [comment]
Agree with Shawn that the "footer buttons" should always use the large size.

Seems like we can ignore the full green option row thingy since we're ditching them anyways. Thanks for investigating that Shawn!

Otherwise I think they are looking good 👍

**Comment by narefyev91** [comment]
@shawnborton ohh sorry did not see your comment for Button - i think it was this one change requested:
"If you don't set any props on the button component, just confirming that the default button we would use is the "medium" version at 40px tall?" - i think for now button by default are using 40px tail.

**Comment by shawnborton** [comment]
Got it, I still think default buttons should use 40px but we need to make sure those footer buttons are using the large size.

**Comment by narefyev91** [comment]
> Got it, I still think default buttons should use 40px but we need to make sure those footer buttons are using the large size.

Problem here - that we have more than 150 buttons in the code - and i'm not really sure - which of them should be large. - Does all button with success styling should be large?

**Comment by shawnborton** [comment]
Nope, not all buttons with success styling should be large. If it's easier to just not make the default size change, we can do that.

**Comment by narefyev91** [comment]
> Nope, not all buttons with success styling should be large. If it's easier to just not make the default size change, we can do that.

Yup it will be much easier

**Comment by shawnborton** [comment]
Definitely curious what the rest of the @Expensify/design thinks though. Just because it's easier, doesn't mean it's the right thing to do. It does feel like we're cutting corners here for the sake of not needing to do the extra work to make it done right.

**Comment by narefyev91** [comment]
> Definitely curious what the rest of the @Expensify/design thinks though. Just because it's easier, doesn't mean it's the right thing to do. It does feel like we're cutting corners here for the sake of not needing to do the extra work to make it done right.

@shawnborton Hmm - I'm not talking about extra work - i was mostly saying - if you could provide all screens in which this bottom (or maybe not all time in the bottom position) button should be large - no problems to make them with large props enabled

**Comment by narefyev91** [comment]
@shawnborton what i can do - in code just to find all buttons which do not have enable prop like large, medium, small - which means that those buttons are using default set of height (which was large before) - and set them large=true

**Comment by shawnborton** [comment]
Yup, makes sense - that's why I would love to do an adhoc build to help you capture all of those places. But it might make sense that any place that currently doesn't have the size prop set would need to have size=large added, given that the default is currently large.

**Comment by luacmartins** [comment]
Kicked off an adhoc build - https://github.com/Expensify/App/actions/workflows/testBuild.yml

**Comment by narefyev91** [comment]
> Nice - @narefyev91 I think this is a good idea:
> 
> > in code just to find all buttons which do not have enable prop like large, medium, small - which means that those buttons are using default set of height (which was large before) - and set them large=true
> 
> In case you wanna give that a shot quickly?

yup - done - added by default 40px - and add for all buttons which do not have size specified large=true
Screenshot 2024-03-12 at 20 00 12

**Comment by shawnborton** [comment]
Lovely! Going to spin up another adhoc build, one sec...

**Comment by dannymcclain** [comment]
I think you guys have landed on the best approach to this. Feels the least risky while also setting things up the way we want them to be. Go team!

**Comment by shawnborton** [comment]
This seems to test pretty well to me! I poked around the app and everything looks normal. 

One tiny tiny request - can you replace our 20x20 QR code icon with this one? [qrcode.svg.zip](https://github.com/Expensify/App/files/14577699/qrcode.svg.zip)

cc @Expensify/design - I think the one in product is an old export that doesn't have any negative space around the icon shape, so it feels beefy compared to what's in Figma:

CleanShot 2024-03-12 at 15 29 37@2x

Figma for reference: 
CleanShot 2024-03-12 at 15 30 11@2x

**Review by luacmartins — COMMENTED** [review]
Just missing Shawn's comment and resolving conflicts

**Comment by narefyev91** [comment]
@luacmartins @shawnborton conflicts fixed, icon is updated
Screenshot 2024-03-13 at 09 55 18

**Comment by shawnborton** [comment]
Looks great, thanks!

**Comment by shawnborton** [comment]
In the PR screenshots, can we be sure to include all of the various button types? So including split buttons, buttons that have an icon to the right and not the left, buttons with no icons, etc. Thanks!

**Comment by narefyev91** [comment]
> In the PR screenshots, can we be sure to include all of the various button types? So including split buttons, buttons that have an icon to the right and not the left, buttons with no icons, etc. Thanks!

I will post one global here for all possible cases:
Screenshot 2024-03-13 at 16 44 55

**Comment by shawnborton** [comment]
Nice, that feels pretty good to me.

**Comment by dannymcclain** [comment]
Same!

File: assets/images/qrcode.svg
Comment (shawnborton) [inline assets/images/qrcode.svg:1]: How did this one become 150x150? It should remain as 20x20. Can you try again with this? 
[qrcode.svg.zip](https://github.com/Expensify/App/files/14590413/qrcode.svg.zip)

Comment (narefyev91) [reply assets/images/qrcode.svg:1]: it's 20x20 in codebase - not sure why github shows it like this

**Review by luacmartins — APPROVED** [review]

**Pull request merged**
